src/common/libflux/fripp.c

```diff
@@ -0,0 +1,438 @@
+
+    return 0;
+}
+
+static void metric_destroy (void **item)
+{
+    if (item) {
+        free (*item);
+        *item = NULL;
+    }
+}
+
+static void timer_cb (flux_reactor_t *r, 
+                      flux_watcher_t *w, 
+                      int revents, 
+                      void *arg)
+{
+    struct fripp_ctx *ctx = arg;
+    const char *name;
+    struct metric *m;
+
+    int rc = 0;
+
+    if (!ctx->buf[0] && zhashx_size (ctx->metrics) == 0) {
+        flux_watcher_stop (ctx->w);
+        return;
+    }
+
+    FOREACH_ZHASHX (ctx->metrics, name, m) {
+        if ((m->type == BRUBECK_COUNTER || m->type == BRUBECK_GAUGE)
+                && m->cur.l == m->prev.l) {
+            zlist_append (ctx->done, (void *) name);
+            continue;
+        }
+        if (m->type == BRUBECK_TIMER && m->cur.d == m->prev.d) {
+            zlist_append (ctx->done, (void *) name);
+            continue;
+        }
+
+        m->prev.l = m->cur.l;
+
+        switch (m->type) {
+            case BRUBECK_COUNTER:
+                rc = fripp_packet_appendf (ctx, "%s:%zd|C\n", name, m->cur.l);
+                break;
+            case BRUBECK_GAUGE:
+                rc = fripp_packet_appendf (ctx, "%s:%zd|g\n", name, m->cur.l);
+                break;
+            case BRUBECK_TIMER:
+                rc = fripp_packet_appendf (ctx, "%s:%lf|ms\n", name, m->cur.d);
+                break;
+            default:
+                break;    
+        }
+
+        if (rc) {
+            fprintf (stderr, "fripp timer_cb: %s\n", strerror (errno));
+            return;
+        }
+    }
+
+    fripp_send_metrics (ctx);
+
+    char *s;
+    while ((s = zlist_pop (ctx->done)))
+        zhashx_delete (ctx->metrics, s);
+        
+    memset (ctx->buf, 0, ctx->buf_len);
+
+    ctx->tail = 0;
+}
+
+void fripp_set_agg_period (struct fripp_ctx *ctx, double period)
+{
+    if (period <= 0) {
+        flux_watcher_stop (ctx->w);
+        ctx->period = 0;
+        return;
+    }
+
+    ctx->period = period;
+    double after = flux_watcher_next_wakeup (ctx->w) - (double) time (NULL);
+
+    flux_timer_watcher_reset (ctx->w, after, ctx->period);
+}
+
+void fripp_ctx_destroy (void *arg)
+{
+    struct fripp_ctx *ctx = arg;
+
+    if (ctx->w) {
+        flux_watcher_stop (ctx->w);
+        flux_watcher_destroy (ctx->w);
+    }
+    if (ctx->metrics) {
+        zhashx_purge (ctx->metrics);
+        zhashx_destroy (&ctx->metrics);
+    }
+    if (ctx->done) {
+        zlist_destroy (&ctx->done);
+    }
+    close (ctx->sock);
+    free (ctx->buf);
+    free (ctx);
+}
+
+struct fripp_ctx *fripp_ctx_create (flux_t *h)
+{
+    struct fripp_ctx *ctx;
+    char *uri, *port_s = NULL, *host = NULL;
+
+    if (!(ctx = calloc (1, sizeof (*ctx)))) {
+        flux_log_error (h, "fripp_ctx_create");
+        goto error;
+    }
+    if (!(uri = getenv ("FLUX_FRIPP_STATSD"))) {
+        flux_log_error (h, "FLUX_FRIPP_STATSD env var not set");
+        goto error;
+    }
```

Comment:
Currently an error is logged if FLUX_FRIPP_STATSD is not set.  Moreover, it's logged for every metric, every time a stats function is called.  FRIPP should be silent if stats collection is not enabled.

Suggestion:  add a function for testing whether stats are enabled:
```c
bool fripp_enabled (flux_t *h, const char *metric);
```
This function can immediately return false if the environment variable is not set, else true.  Later we can use `metric` to test whether a particular metric prefix is enabled, but for now it could just be ignored and set to NULL by callers.

Then call it early to reduce overhead when stats collection is disabled.  For example, in the `flux_stats_*()` functions, return immediately, before trying to access/create context if `fripp_enabled(name)` return false.